superdesk/storage/amazon_media_storage.py

```diff
@@ -84,63 +83,35 @@ class AmazonMediaStorage(MediaStorage):
 
     def __init__(self, app=None):
         super().__init__(app)
-        username, api_key = self.read_from_config()
         self.client = boto3.client(
             's3',
-            aws_access_key_id=username,
-            aws_secret_access_key=api_key,
-            region_name=self.region,
+            aws_access_key_id=self.app.config['AMAZON_ACCESS_KEY_ID'],
+            aws_secret_access_key=self.app.config['AMAZON_SECRET_ACCESS_KEY'],
+            region_name=self.app.config.get('AMAZON_REGION'),
             config=Config(signature_version='s3v4'),
         )
         self.user_metadata_header = 'x-amz-meta-'
 
     def url_for_media(self, media_id, content_type=None):
```

Comment:
seems like this could be defined now on some base class we could use for both amazon and gridfs storage,
same for `url_for_download`